server/src/test/java/org/opensearch/cluster/service/MasterServiceTests.java

```diff
@@ -691,6 +691,266 @@ public void clusterStateProcessed(String source, ClusterState oldState, ClusterS
+            private AtomicInteger published = new AtomicInteger();
+
+            @Override
+            public ClusterTasksResult<Task> execute(ClusterState currentState, List<Task> tasks) throws Exception {
+                latch.countDown();
+                barrier.await();
+                return ClusterTasksResult.<Task>builder().successes(tasks).build(currentState);
+            }
+
+            @Override
+            public String getMasterThrottlingKey() {
+                return taskName;
+            }
+
+            @Override
+            public void clusterStatePublished(ClusterChangedEvent clusterChangedEvent) {
+                published.incrementAndGet();
+            }
+        }
+
+        MasterService masterService = createClusterManagerService(true);
+        masterService.masterTaskThrottler.updateThrottlingLimit(taskName, throttlingLimit);
+
+        final ClusterStateTaskListener listener = new ClusterStateTaskListener() {
+            @Override
+            public void onFailure(String source, Exception e) {}
+
+            @Override
+            public void clusterStateProcessed(String source, ClusterState oldState, ClusterState newState) {}
+        };
+
+        TaskExecutor executor = new TaskExecutor();
+        // submit one task which will be execution, post that will submit throttlingLimit tasks.
+        try {
+            masterService.submitStateUpdateTask(
+                taskName,
+                new Task(taskId++),
+                ClusterStateTaskConfig.build(randomFrom(Priority.values())),
+                executor,
+                listener
+            );
+        } catch (Exception e) {
+            throw new AssertionError(e);
+        }
+        // wait till task enter in execution.
+        latch.await();
+
+        for (int i = 0; i < throttlingLimit; i++) {
+            try {
+                masterService.submitStateUpdateTask(
+                    taskName,
+                    new Task(taskId++),
+                    ClusterStateTaskConfig.build(randomFrom(Priority.values())),
+                    executor,
+                    listener
+                );
+            } catch (Exception e) {
+                throw new AssertionError(e);
+            }
+        }
+
+        // we have one task in execution and tasks in queue so next task should throttled.
+        final AtomicReference<MasterTaskThrottlingException> assertionRef = new AtomicReference<>();
+        try {
+            masterService.submitStateUpdateTask(
+                taskName,
+                new Task(taskId++),
+                ClusterStateTaskConfig.build(randomFrom(Priority.values())),
+                executor,
+                listener
+            );
+        } catch (MasterTaskThrottlingException e) {
+            assertionRef.set(e);
+        }
+        assertNotNull(assertionRef.get());
+        masterService.close();
+    }
+
+    public void testThrottlingForMultipleTaskTypes() throws InterruptedException {
```

Comment:
what is this test trying to accomplish ? Is it just the independence of the multiple task types ? Else it is just trying to do a loop on `testThrottlingForTaskSubmission`  test .

Reply:
Yes independence of multiple task types, as different tasks are getting throttled independently with different limits.